app/pages/UcsfClientEligibilityPage/UcsfClientEligibilityPage.tsx

```diff
@@ -124,23 +111,42 @@ const ClientEligibilities = ({ rawEligibilityData, resourceSlug }: {
 };
 
 const Page = () => {
+  const [selectedEligibilities, setSelectedEligibilities] = useState<SelectedEligibilities>({});
+
   interface LocationState {
     selectedResourceSlug: string;
+    selectedEligibilityNames: string[];
   }
 
   const history = useHistory<LocationState>();
   const { state } = history.location;
   const selectedResourceSlug = state && state.selectedResourceSlug;
+  const resourceEligibilityGroups = eligibilityMap[selectedResourceSlug];
+
   const goToServiceTypePage = (slug: string) => {
-    history.push('/service-type', { selectedResourceSlug: slug });
+    const flattenedEligibilityGroups = resourceEligibilityGroups.reduce<Eligibility[]>(
+      (previousValue, currentValue) => previousValue.concat(currentValue.eligibilities),
+      [],
+    );
+
+    const selectedEligibilityNames = flattenedEligibilityGroups.reduce<string[]>((
+      result,
+      eligibility,
+    ) => {
+      if (selectedEligibilities[eligibility.checkedId] && !eligibility.isSeeAll) {
+        return [...result, eligibility.name];
+      }
+      return result;
+    }, []);
```

Comment:
Similar to my `flat()` suggestion, could this be expressed with `filter()`?

Reply:
Since I only want to return the eligibility names, using filter would have required using map as well (I think). But I think I found a better solution in the `flatMap` method. See my comment above